src/main/java/org/opensearch/security/privileges/SystemIndexAccessEvaluator.java

```diff
@@ -307,11 +311,37 @@ private void evaluateSystemIndicesAccess(
                         );
                     }
                     presponse.allowed = false;
+                    presponse.addMissingPrivileges(action);
                     presponse.markComplete();
                     return;
                 }
         }
 
+        if (user instanceof PluginUser) {
```

Comment:
This doesn't seem necessary, if the user principle that represents a plugin is trustworthy the lookup for allowed system indices should return an empty set for a normal user.  Lets treat all users the same in code, but allow there configuration data to be populated in a way that distinguishes them well.

By special casing for a plugin user vs normal user create a huge potential for bugs down the line where these permissions are not accounted correctly that would result in a CVE.  Note; If there are performance concerns then we need to revisit how these data structures are build to be O(1) lookup.

Reply:
That makes sense, I just pushed a change gets rid of the need for the subclass.

I plan to use a username convention for these special "plugin users". The name will be `plugin:<pluginCanonicalClassName>`. I'm choosing this because `:` is [restricted from usernames](https://github.com/opensearch-project/security/blob/e7a79a861db204e021f8b6b5e66e9d6927f45178/src/main/java/org/opensearch/security/user/UserService.java#L92-L94), so this will provide assurance that this is a special kind of system username and not a user that could have been created through any other means.